Sort JSON Data

hmajid2301 opened the pull request:
#### Short description of what this resolves:

Sort the JSON (settings) files before they are uploaded.


#### Changes proposed in this pull request:

- If the sort setting is enabled
- Sort every JSON file before upload


**Fixes**: #998 

#### How Has This Been Tested?
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, tests ran to see how -->
<!--- your change affects other areas of the code, etc. -->

As per my slack messages I have been unable to test it locally.
Perhaps someone can test it on their machine (locally). 

#### Screenshots (if appropriate):


#### Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] I have read the [contribution](https://github.com/shanalikhan/code-settings-sync/blob/master/CONTRIBUTING.md#setup-extension-locally) guidelines.
- [x] My change requires a change to the documentation and GitHub Wiki.
- [ ] I have updated the documentation and Wiki accordingly.

Commit:
work

Commit:
work

Commit:
work

Commit:
Merge branch 'v3.4.4' of https://github.com/shanalikhan/code-settings-sync into feature/#998

Commit:
Updated for new changes

Commit:
Small Fix

Commit:
Fixed Linting Errors

Fixed linting errors caught when using
`npm run tslint-check`

shanalikhan commented on package.json:
```diff
@@ -134,6 +134,11 @@
           "default": true,
           "description": "%ext.config.removeExtensions%"
         },
+        "sync.sortAlphabetically": {
```
can you put it as global settings instead of env. settings.

hmajid2301 replied:
How can i do that ?

hmajid2301 replied:
What about this comment ?

shanalikhan replied:
`sortAlphabetically` needs to be moved into 

https://github.com/shanalikhan/code-settings-sync#global-settings

In this file
https://github.com/shanalikhan/code-settings-sync/blob/master/src/models/customConfig.model.ts

shanalikhan reviewed (changes requested):
Hello @hmajid2301 !

Sorry i was away from last month. I'm going to review this PR. Pls complete the changes so i can merge into next release branch

nandalopes commented on README.md:
```diff
@@ -154,6 +154,7 @@ You can customize the settings in gist settings like:
     "sync.forceDownload": false,
     "sync.forceUpload": false,
     "sync.quietSync": false,
+    "sync.sortAlphabetically", false,
```
Change comma to colon

"sync.sortAlphabetically": false,

hmajid2301 committed:
fix: Typo in README

change , to : in the README for alphabetical sort.

shanalikhan committed:
Merge branch 'v3.4.4' into feature/#998

shanalikhan commented:
Can you fix this or i can close this PR.
Thanks for the work, the only thing remaining is to put the configuration in Global Settings of settings sync

Here : https://github.com/shanalikhan/code-settings-sync#global-settings

Instead of the current place of configuration you have added.

Pls fix and let me know so i can add in upcoming release

hmajid2301 commented:
Ok I will fix the issues

Commit:
fix: Move config to correct file location

Moved the sort alphabetically config option to the
correct file. Moved it from `package.json`.

shanalikhan committed:
Merge branch 'v3.4.4' into feature/#998
